tuning/sft_trainer.py

```diff
@@ -461,12 +461,46 @@ def parse_arguments(parser, json_config=None):
     )
 
 
-def main(**kwargs):  # pylint: disable=unused-argument
-    logger = logging.get_logger("__main__")
+def set_log_level(parsed_training_args):
+    """Set log level of python native logger and TF logger via argument from CLI or env variable.
+
+    Args:
+        parsed_training_args
+            Training arguments for training model.
+    """
+
+    # Clear any existing handlers if necessary
+    for handler in logging.root.handlers[:]:
+        logging.root.removeHandler(handler)
+
+    # Configure Python native logger log level
+    # If CLI arg is passed, assign same log level to python native logger
+    if parsed_training_args.log_level != "passive":
+        logging.basicConfig(level=parsed_training_args.log_level.upper())
+    else:
+        # Assign value of either env var LOG_LEVEL or warning
+        LOGLEVEL = os.environ.get("LOG_LEVEL", "WARNING").upper()
+        logging.basicConfig(level=LOGLEVEL)
+        train_logger = logging.getLogger()
+
+    # Check if env var TRANSFORMERS_VERBOSITY is not set.
+    # Else if env var is already set then, log level of transformers is automatically set.
+    if os.environ.get("TRANSFORMERS_VERBOSITY") is None:
```

Comment:
nit: can do 
```suggestion
   if not os.environ.get("TRANSFORMERS_VERBOSITY"):
```

Reply:
Although do we even need the code block below? What use case is it covering? Do we have to check that TRANFORMERS_VERBOSITY at all?

is this trying to do
```py
if os.getenv("TRANFORMERS_VERBOSITY"):
   parsed_training_args.log_level = os.getenv("TRANFORMERS_VERBOSITY").lower()
```

Reply:
I don't think we are required to check "TRANFORMERS_VERBOSITY". Hence I haven't added this code in final commit.